db/queries/scores/queries.py

```diff
@@ -0,0 +1,63 @@
+"""Queries which are performed on the `scores` table.
+
+Joins allowed.
+"""
+import json
+from typing import Dict
+from typing import List
+
+from db import db
+from db.models.score import JustScore
+from db.queries.assessment_records._helpers import derive_values_from_json
+from db.schemas import JustScoreMetadata
+from sqlalchemy import func
+from sqlalchemy import select
+from sqlalchemy.orm import defer
+
+
+def get_just_score_for_application_sub_crit(
+    application_id: str, sub_criteria_id: str
+) -> Dict:
+    """get_just_score_for_application_sub_crit executes a query on scores
+    which returns the most recent score for the given application_id and 
+    sub_criteria_id.
+
+    :param application_id: The stringified application UUID.
+    :param sub_criteria_id: The stringified sub_criteria UUID.
+    :return: dictionary.
+    """
+    stmt = select(JustScore).where(
+        JustScore.application_id == application_id,
+        JustScore.sub_criteria_id == sub_criteria_id
+        ).order_by(JustScore.timestamp.desc())
```

Comment:
because I am using method scalar() this will always return one result :) and in terms of sorting, I'm not 100% sure if we're always going to be sorting by either DESC or ASC but looks very cool!! :D - something to consider when we know more about score histories etc... ;)

Reply:
Yes, but SQLA will just take a single value from whatever is returned by postgres - there will still be quite a heavy load on the database. Do check the sql output in pytest and see what statement is emitted here.

Reply:
but yes scalar is better then one, good spot :)

Reply:
and the index can be used regardless of ASC or DESC, but you if you know then you can specify to make the index super small, if not it doesn't matter....**tl;dr** add the index